src/api/MainSolver.cc

```diff
@@ -333,14 +334,24 @@ sstat MainSolver::check() {
         StopWatch sw(query_timer);
     }
     if (isLastFrameUnsat()) { return s_False; }
-    sstat rval = simplifyFormulas();
+    sstat rval;
+    try {
+        rval = simplifyFormulas();
+    } catch (opensmt::NonLinException const & error) {
+        reasonUnknown = error.what();
+        rval = s_Undef;
+        return rval;
+    }
 
     if (config.dump_query()) printCurrentAssertionsAsQuery();
 
     if (rval == s_Undef) {
         try {
             rval = solve();
-        } catch (std::overflow_error const & error) { rval = s_Error; }
+        } catch (std::overflow_error const & error) { rval = s_Error; } catch (opensmt::NonLinException const & error) {
+            reasonUnknown = error.what();
+            rval = s_Undef;
+        }
```

Comment:
```suggestion
        } catch (std::overflow_error const & error) { rval = s_Error; } catch (opensmt::NonLinException const & error) {
            reasonUnknown = error.what();
            return  s_Undef;
        }
```

And please format the catch statements.

Reply:
Fixed

Reply:
I'm not sure what to do about it, this is how Formatter requires us to do it, it complaints otherwise